crates/api/src/fetch.rs

```diff
@@ -0,0 +1,28 @@
+//! Kitsune2 fetch queue types.
+
+use std::sync::Arc;
+
+use crate::{AgentId, OpId};
+
+/// Trait for implementing a fetch queue for managing ops
+/// to be fetched from other agents.
+pub trait FetchQueueT: 'static + Send + Sync {
+    /// Add op ids to be fetched to the queue.
+    fn add_ops(&mut self, op_list: Vec<OpId>, source: AgentId);
+
+    /// Get a list of op ids to be fetched.
+    fn get_ops_to_fetch(&self) -> Vec<OpId>;
+
+    /// Get a random agent that holds the ops to be fetched.
+    fn get_random_source(&self) -> Option<AgentId>;
```

Comment:
I think we should start with *only* `add_ops`. Gossip may eventually need some extra view into how overworked the fetch queue is, but we don't really know what that needs to look like yet, and also it may not need that. To start, I think it'd be fine if it redundantly added ops again, and that was just a no-op.

Reply:
I understand now. There's just the fetch queue that manages state and fetching, and all consumers do is add op lists with agent to it. How will the fetch response queue or its owning struct remove ops that have been fetched?

Reply:
I realize that that is an optimization and depends on how the gossip protocol will work.

Reply:
My intention with the tx module is that other modules (such as this FetchQueue module) can register sub-protocol handlers. So the fetch module would send op hashes to remotes, those remotes would send the op data back, and the fetch queue module would receive it and be able to remove the item from the queue.

Looking at it this way, perhaps the response queue actually needs to be managed by this module as well so we don't have to split the protocol between multiple modules.